src/hardware/input/mouse_manymouse.h

```diff
@@ -103,7 +103,9 @@ class ManyMouseGlue final {
 	std::vector<int> rel_x = {}; // not yet reported accumulated movements
 	std::vector<int> rel_y = {};
 
-	static constexpr uint8_t max_buttons  = 3;
+	static constexpr uint8_t max_buttons = 3;
+	static_assert(max_buttons <= static_cast<uint8_t>(MouseButtonId::Last));
```

Comment:
Minor but `max_buttons` could theoretically go up to `MouseButtonId::Last + 1` right?  You've got 0-4 as valid buttons so 5 total buttons with `MouseButtonId::Last` being mapped to 4 as the last valid index.

Reply:
With mice mapped using ManyMouse we support up to 3 buttons. Yes, the library supports more, but I am not sure how it maps the extra buttons, which ID it uses (if I remember correctly, this is not straightforward). Since the only real-life use case I know are Settlers I / II games, I didn’t bother to spend to much effort on this.

Reply:
Yeah, I was just talking about the `static_assert` being slightly misleading.  Not suggesting to change the actual constant here or implement new functionality.

Reply:
The many-mouse feature is a subset feature of the overall mouse system, and it currently only handles a subset of buttons.

Many-mouse also uses raw integers of the button IDs, which we cast over to the new enum - so this check just makes sure those casts are always going to map over to our enum types.

@weirddan455, maybe we could rename `max_buttons` to clarify that this is a many-mouse limit? 

Something like:

```c++
static constexpr auto many_mouse_last_button = 
             static_cast<uint8_t>(MouseButtonId::Middle);
```

Reply:
@kcgen Naming seems to be the disconnect here.  Are we talking about the maximum number of buttons or the index of the last valid button?  If you just re-name it like you suggested, the constant changes from 3 to 2 and you've got an off 1 by error.

Reply:
If we are indeed talking about maximum number of buttons, I would suggest adding to the end of the enum:

```cpp
NumButtons = 5,
```

Then the static assert becomes the following to avoid confusion:

```cpp
static_assert(max_buttons <= static_cast<uint8_t>(MouseButtonId::NumButtons));
```

Reply:
Reworked as a separate clarification commit. (See 2nd commit).